Rubberduck.Main/ComClientLibrary/UnitTesting/Mocks/ComMocked.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using NLog;
+
+namespace Rubberduck.ComClientLibrary.UnitTesting.Mocks
+{
+    public class ComMocked : ICustomQueryInterface
+    {
+        private static readonly ILogger Logger = LogManager.GetCurrentClassLogger();
+        private readonly object _target;
+        private readonly IEnumerable<Type> _supportedTypes;
+
+        internal ComMocked(object target, IEnumerable<Type> supportedTypes)
+        {
+            _target = target;
+            _supportedTypes = supportedTypes;
+        }
+
+        private static readonly Guid iidIDispatch = new Guid("{00020400-0000-0000-C000-000000000046}");
+        public CustomQueryInterfaceResult GetInterface(ref Guid iid, out IntPtr ppv)
+        {
+            try
+            {
+                var result = IntPtr.Zero;
+                var searchIid = iid;
+
+                if (iid == iidIDispatch)
+                {
+                    // TODO: find a better way to get IDispatch - it is not possible to return
+                    // a IDispatch because it is in a object hierarchy where the parent object
+                    // is not COM visible. Returning IUnknown seems to work because of the happy
+                    // accident that the IDispatch's v-table usually follows IUnknown's v-table.
```

Comment:
If you have an `IUnknown`, can't you just call `QueryInterface` and _ask_ for its `IDispatch` pointer?

Reply:
As mentioned in chat but documenting for posterity - trying to get `IDispatch` from a `QueryInterface` (or `Marshal.GetIDispatchForObject`) will yield the same error because the Moq.Mock proxy itself is not COM-visible even though it has COM-visible interfaces.